gulp-core-build-typescript/src/TypeScriptTask.ts

```diff
@@ -140,7 +146,14 @@ export class TypeScriptTask extends GulpTask<ITypeScriptTaskConfig> {
       .pipe(sourcemaps.init())
       .pipe(ts(tsProject, undefined, this.taskConfig.reporter));
 
-    allStreams.push(tsResult.js
+    // tslint:disable-next-line:typedef
+    const jsResult = (this.taskConfig.removeCommentsFromTypeScript
+      ? tsResult.js.pipe(require('gulp-decomment')({
+        space: true /* leave this on for sourcemaps */
```

Comment:
Will the sourcemaps contain the original TS files? Because that kinda defeats the purpose here.